Fix: Default radio button for Mock Implementation endpoints when adding Mock endpoint.

iff-sal opened the pull request:
Resolves wso2/api-manager#3680

**What was the problem?**
When a user added a "Mock Implementation" for an API endpoint, the UI would incorrectly select the "Dynamic Endpoints" radio button by default on the subsequent configuration page.
This was due to the createEndpointConfig utility function lacking a specific case to handle the 'INLINE' endpoint type. As a result, it fell through to the default case, generating a configuration that the UI interpreted as a "Dynamic Endpoint".

**How was this fixed?**
This pull request implements a complete fix by addressing the issue at all three relevant points in the code, ensuring 'INLINE' endpoints are handled correctly and consistently.
Updated the Reducer (Endpoints.jsx): The reducer's 'endpointImplementationType' case was modified to explicitly handle an endpointType of 'INLINE'. It now correctly sets endpointImplementationType: 'INLINE' in the state and triggers the mock script generation, resolving the primary bug.
Added 'INLINE' Case (endpointUtils.js): A new case for 'INLINE' was added to the createEndpointConfig function. This ensures a valid, prototyped configuration is generated, preventing the system from falling back to a 'default' config.
Removed Workaround (EndpointOverview.jsx): The inconsistent logic that mapped 'INLINE' to 'prototyped' within the component has been removed. The component now calls createEndpointConfig with the correct 'INLINE' type directly, centralizing the configuration logic.

**How to test this:**
Log in to the Publisher portal and create an API without endpoints.
navigate to an API's Endpoints section under API Configurations.
Under the Mock Implementation card, click the Add button.
On the endpoint configuration page, verify that the "Mock Implementation" radio button is now correctly selected by default (and not "Dynamic Endpoints").

RusJaI commented on portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/Endpoints.jsx:
```diff
@@ -198,6 +198,12 @@ function Endpoints(props) {
                     }
                     return { ...initState, endpointConfig: config, endpointImplementationType: 'ENDPOINT' };
                 }
+                if (endpointType === 'INLINE') {
```
The codebase intends to branch on endpointImplementationType. Adding a logic to handle behavior specifically for the INLINE type would duplicate that logic and could lead to inconsistencies if the two fields ever differ. Let’s avoid adding this special case here.

RusJaI commented on portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:
```diff
@@ -564,7 +564,7 @@ function EndpointOverview(props) {
         const selectedKey = typeChangeConfirmation.type || value;
         if (selectedKey === 'INLINE' || selectedKey === 'MOCKED_OAS') {
-            const tmpConfig = createEndpointConfig('prototyped');
+            const tmpConfig = createEndpointConfig(selectedKey);
```
Changing the endpoint type is currently working as intended. If you select the “Mock Implementation” radio button after choosing another endpoint type, the UI correctly navigates to the correct view.

RusJaI commented on portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/endpointUtils.js:
```diff
@@ -190,6 +190,12 @@ function createEndpointConfig(endpointType) {
             tmpEndpointConfig.production_endpoints = { config: null, url: 'http://localhost' };
             tmpEndpointConfig.sandbox_endpoints = { config: null, url: 'http://localhost' };
             break;
+        case 'INLINE': // mock implementation is functionally prototype.
```
The intended endpointType for this scenario should be 'prototyped'. Therefore, the fix should involve updating the code to reflect this behavior, rather than adding a special case to pass the incoming value.
Which means, the actual issue arises before coming to this level.

RusJaI reviewed (commented):
Thank you for taking the time to work on this fix! 
I’ve added a few comments to the PR. The fix should ideally be applied at a higher level in the code than you have currently focused.

iff-sal committed:
fix: Set default correct endpoint in radio button for Mock implementation

iff-sal commented:
@RusJaI I implemented your suggestion. Instead of changing the low-level logic and creating duplicates, I just fixed it at a higher level by passing the correct endpointImplementationType from the apiReducer. This ensures the correct props are passed down, and the UI defaults to the right radio button. Thanks for the review!

https://github.com/user-attachments/assets/924f94c3-d21a-40a8-a5c0-e5494f999753

RusJaI commented on portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/Endpoints.jsx:
```suggestion
                const config = createEndpointConfig(endpointType);
                if (endpointType === 'INLINE' || endpointType === 'MOCKED_OAS') {
                    const tmpconfig = createEndpointConfig('prototyped');
                    if (implementationType === 'mock') {
                        api.generateMockScripts(api.id).then((res) => { // generates mock/sample payloads
                            setSwagger(res.obj);
                        });
                        return { ...initState, endpointConfig: tmpconfig, endpointImplementationType: endpointType };
                    }
                    return { ...initState, endpointConfig: tmpconfig, endpointImplementationType: 'ENDPOINT' };
                }
                return { ...initState, endpointConfig: config };
            }
```

RusJaI reviewed (changes requested):
You’ve identified the correct place in the code. Instead of adding a new if clause, let’s reuse the existing one as suggested. Also noted that your current fix affects the generation of mock scripts, which will be resolved with the suggested approach

iff-sal committed:
Refactor endpoint implementation handling logic

use the existing if condition to check the logic

iff-sal commented:
Hi @RusJaI, Thanks for the review and patience. I got busy with university exam stuff and the exams are started, so didnt had lot of time to give my full focus to this issue. I implemented your suggestion of implementing the logic in the existing if statement and committed it. Again thanks for the feedback.
